akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -175,7 +167,15 @@ import akka.japi.function.{ Function ⇒ JFunction }
           extractShardId)
       }
 
-    typeNames.put(typeKey.name, typeKey.asInstanceOf[EntityTypeKeyImpl[A]].messageClassName)
+    val messageClassName = typeKey.asInstanceOf[EntityTypeKeyImpl[A]].messageClassName
+
+    val typeNames = if (settings.shouldHostShard(cluster)) regions else proxies
+
+    typeNames.putIfAbsent(typeKey.name, messageClassName) match {
+      case spawnedMessageClassName: String if messageClassName != spawnedMessageClassName ⇒
+        throw new IllegalArgumentException(s"${typeKey.name} already spawned for $spawnedMessageClassName")
```

Comment:
even though not consistent everywhere we try to use brackets around such fields in exception and log messages:

```
s"[${typeKey.name}] already spawned for [$spawnedMessageClassName]"
```